src/Router.tsx

```diff
@@ -40,6 +42,68 @@ export const RouterInner = () => {
   const { sideMenuOpen, sideMenuMinimised, setContainerRefs } = useUi();
   const { t } = useTranslation('base');
 
+  const [urlNetworkExists, setUrlNetworkExists] = useState<boolean>(false);
+  const [urlNetwork, setUrlNetwork] = useState<any>('');
+  const [urlLngExists, setUrlLngExists] = useState<boolean>(false);
+  const [urlLng, setUrlLng] = useState<any>('');
```

Comment:
None of these state items are needed. The active `network` and `switchNetwork` function are stored in the `API` context, and the active locale is  managed by `i18n.resolvedLanguage` / `i18n.changeLanguage()`. 

Use these directly. If the URL vars do not exist, nothing needs to be done.